app/controllers/posts_controller.rb

```diff
@@ -1,52 +1,51 @@
 class PostsController < ApplicationController
   before_action :authenticate_account!, except:  [ :index, :show ]
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
-  before_action :auth_subscriber, only: [:new]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :increment_view_count]
+  before_action :find_my_communities, only: [:new, :create, :edit, :update]
   before_action :community_list
   def index
     @posts = Post.all
   end
 
   def show
+    increment_view_count
     @comment = Comment.new
   end
 
   def new
-    @community = Community.find(params[:community_id])
-    @post = Post.new 
+    @post = Post.new
     @drafts = Post.order(created_at: :desc).page(params[:page]).per 5
   end
 
   def create
+    @drafts = Post.all
```

Comment:
Don't use the .all api because it pulls all the data from db. What if we have a lot of data? then it slows down which is not ideal so restrict the limit to 20-30 records always

Reply:
Resolved it